internal/venice-common/src/main/resources/avro/KafkaMessageEnvelope/v11/KafkaMessageEnvelope.avsc

```diff
@@ -0,0 +1,319 @@
+                    }
+                  ]
+                }, {
+                  "name": "EndOfIncrementalPush",
+                  "doc": "This ControlMessage is sent per partition by each offline incremental push job, once per partition, at the end of a incremental push",
+                  "type": "record",
+                  "fields": [
+                    {
+                      "name": "version",
+                      "doc": "The version of current incremental push. Each incremental push is associated with a version. Both 'StartOfIncrementalPush' control message and 'EndOfIncrementalPush' contain version info so they can be paired to each other.",
+                      "type": "string"
+                    }
+                  ]
+                }, {
+                  "name": "TopicSwitch",
+                  "doc": "This ControlMessage is sent by the Controller, once per partition; it will only be used in leader/follower state transition model; this control message will indicate the leader to switch to a new source topic and start consuming from offset with a specific timestamp.",
+                  "type": "record",
+                  "fields": [
+                    {
+                      "name": "sourceKafkaServers",
+                      "doc": "A list of Kafka bootstrap servers URLs where the new source topic exists; currently there will be only one URL in the list, but the list opens up the possibility for leader to consume from different fabrics in active-active replication mode.",
+                      "type": {
+                        "type": "array",
+                        "items": "string"
+                      }
+                    }, {
+                      "name": "sourceTopicName",
+                      "doc": "Name of new the source topic.",
+                      "type": "string"
+                    }, {
+                      "name": "rewindStartTimestamp",
+                      "doc": "The creation time of this control message in parent controller minus the rewind time of the corresponding store; leaders in different fabrics will get the offset of the source topic by the same start timestamp and start consuming from there; if timestamp is 0, leader will start consuming from the beginning of the source topic.",
+                      "type": "long"
+                    }
+                  ]
+                }, {
+                  "name": "VersionSwap",
+                  "doc": "This controlMessage is written to the real-time topic by the controller or to the store-version topic by the current version's leader server. It can be used to let current version and future version synchronize on a specific point for all regions' real-time topics, to guarantee there is only one store version producing to change capture topic all the time. It can also be used by the consumer client to switch to another store-version topic and filter messages that have a lower watermark than the one dictated by the leader.",
```

Comment:
One last thing to add here.  It was suggested at one point if we should have a boolean for if this is a repush or not.  We were originally kind of wishy washy on it, but now I think we should add it.  I spoke with some folks in search and it sounds like they'd actually like some kind of behavior for when version swap happens, but probably NOT when repush happens (and the underlying data is the same).  We may or may not make that work for them in the future, but, it would seem to me that there are cases where that might make some kind of sense (just spitballing, it might be something we want davinci to act on when we start doing repushes in place of Kafka compaction, we might actually want davinci to skip reboostrapping itself in that kind of scenario?  I'm speculating but it seems like that'd be a useful property).

Reply:
Not sure if it is good for 'Version Swap' control message to carry this version specific information, or it should be saved in `Version` for that version?